corehq/apps/app_manager/models.py

```diff
@@ -4160,17 +4160,27 @@ def get_app(domain, app_id, wrap_cls=None, latest=False):
             parent_app_id = original_app['_id']
             min_version = -1
 
-        latest_app = get_db().view('app_manager/applications',
-            startkey=['^ReleasedApplications', domain, parent_app_id, {}],
-            endkey=['^ReleasedApplications', domain, parent_app_id, min_version],
-            limit=1,
-            descending=True,
-            include_docs=True
-        ).one()
+        if target and target == 'build':
+            # get latest-build regardless of star
+            latest_app = get_db().view('app_manager/saved_app',
+                                       startkey=[domain, parent_app_id, {}],
+                                       endkey=[domain, parent_app_id],
+                                       descending=True,
+                                       limit=1,
+                                       include_docs=True).one()
+        else:
+            # get latest starred-build
+            latest_app = get_db().view('app_manager/applications',
+                                       startkey=['^ReleasedApplications', domain, parent_app_id, {}],
+                                       endkey=['^ReleasedApplications', domain, parent_app_id, min_version],
+                                       limit=1,
+                                       descending=True,
+                                       include_docs=True).one()
```

Comment:
Rather than duplicating the entire view query you could just create the star and end keys in the if / else statement